packages/components/src/Checkbox/Checkbox.mdx

```diff
@@ -120,6 +123,57 @@ that the user has lost control of the interface.
   />
 </Playground>
 
+## With an embedded react component
+
+If you require more functionality than a simple string for the label, it is
+possible to pass a child component to the checkbox to use as a label. This
+allows us to embed a link into the checkbox label.
+
+<Playground>
+  {() => {
+    const [checked, setChecked] = useState(false);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    function handleOpenModal(e) {
+      e.preventDefault();
```

Comment:
In most cases, we put the most common scenario that the people can just copy paste. This one, however, needs a bit more context as to why there's a prevent default there. It's not really meant to prevent the default of the link, it's preventing the default of the `label` that holds this link.

```suggestion
      // Prevent the checkbox from being checked
      e.preventDefault();
```